ext/mbstring/libmbfl/filters/mbfilter_cp936.c

```diff
@@ -142,24 +142,13 @@ int mbfl_filt_conv_cp936_wchar(int c, mbfl_convert_filter *filter)
 
 		if (w <= 0) {
 			if (c1 < 0xff && c1 > 0x80 && c >= 0x40 && c < 0xff && c != 0x7f) {
-				w = (c1 - 0x81)*192 + (c - 0x40);
-				if (w >= 0 && w < cp936_ucs_table_size) {
-					w = cp936_ucs_table[w];
-					if (!w)
-						w = MBFL_BAD_INPUT;
-				} else {
-					w = MBFL_BAD_INPUT;
-				}
+				w = cp936_ucs_table[(c1 - 0x81)*192 + (c - 0x40)];
```

Comment:
Might make sense to `ZEND_ASSERT(w < cp936_ucs_table_size)` etc?

Reply:
Good idea. Pushing an update now.